File: perfect_pair server/perfect_pair/src/main/java/com/olduo/last_dance/model/service/AnswerService.java
package com.olduo.last_dance.model.service;

import java.util.List;
import java.util.Map;

import com.olduo.last_dance.model.dto.Answer;

public interface AnswerService {
	/**
     * user의 answer를 등록한다.
     * @param answer
     */
	Boolean addAnswer(Answer answer);
    
    /**
     * answer에 해당하는 Answer을 수정한다.
     * @param answer
     */
    int updateAnswer(Answer answer);
    
    /**
     * qId와 uId에 해당하는 Answer 객체에서 사용자의 answer를 반환한다.
     * @param qId, uId
     * @return answer
     */
    Map selectAnswer(Map map);
    
    /**
     * qId에 해당하는 Answer의 목록을 반환한다.
     * @param userId
     * @return
     */
    List<Map> getAnswerByQuiz(Integer qId);
}

File: app/perfect_pair/data/src/main/java/com/olduo/last_dance/data/datasource/remote/service/AnswerService.kt
package com.olduo.last_dance.data.datasource.remote.service

import com.olduo.last_dance.data.model.AnswerDto
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path

interface AnswerService {
    @POST("answer")
    suspend fun sendAnswer(@Body answerDto: AnswerDto): Response<Boolean>

    @PUT("answer")
    suspend fun updateAnswer(@Body answerDto: AnswerDto): Response<Boolean>

    @GET("answer/{qId}")
    suspend fun getAllAnswers(@Path("qId") qId: Int): Response<List<AnswerDto>>

    @GET("answer/{qId}/{userId}")
    suspend fun getUserAnswer(@Path("qId") qId: Int, @Path("userId") userId: String): Response<AnswerDto>
}